specs/spec_reader.go

```diff
@@ -173,3 +182,34 @@ func NewSpecReader(paths []string) (*SpecReader, error) {
 
 	return reader, nil
 }
+
+// strip yaml comments from the given yaml document by converting to JSON and back :)
+func stripYamlComments(b []byte) ([]byte, error) {
+	const openPlaceholder = "$$$OPEN$$$"
+	const closePlaceholder = "$$$CLOSE$$$"
+
+	// return an error if the yaml already contains our temporary placeholder for env variables by some unlucky coincidence
+	if bytes.Contains(b, []byte(openPlaceholder)) || bytes.Contains(b, []byte(closePlaceholder)) {
+		return nil, fmt.Errorf("%s and %s are reserved words in CloudQuery config", openPlaceholder, closePlaceholder)
+	}
+
+	// replace placeholder variables with valid yaml, otherwise it cannot be parsed
```

Comment:
GitHub wrote their own YAML parser. We should not do it https://github.com/cloudquery/plugin-sdk/issues/480#issuecomment-1404694684